addon/wait-for-promise.ts

```diff
@@ -25,22 +25,26 @@ const PROMISE_WAITER = buildWaiter('promise-waiter');
  *   }
  * }
  */
-export default function waitForPromise<T>(promise: Promise<T>, label?: string): Promise<T> {
+export default function waitForPromise<T, PromiseType extends Promise<T> | RSVPPromise<T>>(
```

Comment:
This looks good. I looked into us potentially using `PromiseLike<T>`, but using that would narrow the return type (it wouldn't include `catch` and `finally`). 

Could we add tests for both variants, native and RSVP?

Reply:
yea!

Reply:
Thank you!